autorest/responder.go

```diff
@@ -153,6 +153,25 @@ func ByClosingIfError() RespondDecorator {
 	}
 }
 
+// ByUnmarshallingBytes returns a RespondDecorator that copies the Bytes returned in the
+// response Body into the value pointed to by v.
+func ByUnmarshallingBytes(v *[]byte) RespondDecorator {
+	return func(r Responder) Responder {
+		return ResponderFunc(func(resp *http.Response) error {
+			err := r.Respond(resp)
+			if err == nil {
+				bytes, errInner := ioutil.ReadAll(resp.Body)
+				if errInner != nil {
+					err = fmt.Errorf("Error occurred reading http.Response#Body - Error = '%v'", errInner)
+				} else {
+					copy(*v, bytes)
```

Comment:
This is going to make a copy of the source bytes read which seems inefficient.  Can't we simply do the following.
```go
v = bytes
```

Reply:
I tried that, but that ended up returning an empty slice - I think because the body's cleaned up during the close?

Reply:
It's true that we call `Close()` on the body however that shouldn't invalidate the slice as the bytes read from the body are placed in a `bytes.Buffer`.  Just to confirm, you did `v = bytes` and not `*v = bytes` yes?

Reply:
Running the tests with this change it appears that's fine - so maybe it was a bug heh 🙃

```
$ envchain azurerm go test -v ./storage/2018-11-09/file/files/ -run="TestPutSmallFile"
=== RUN   TestPutSmallFile
2019/06/19 18:57:01 Testing if Service Principal / Client Certificate is applicable for Authentication..
2019/06/19 18:57:01 Testing if Service Principal / Client Secret is applicable for Authentication..
2019/06/19 18:57:01 Using Service Principal / Client Secret for Authentication
2019/06/19 18:57:29 [DEBUG] Chunk 1 of 1
--- PASS: TestPutSmallFile (75.94s)
    range_put_file_test.go:64: [DEBUG] Creating Top Level File..
    range_put_file_test.go:73: [DEBUG] Uploading File..
    range_put_file_test.go:78: [DEBUG] Deleting Top Level File..
PASS
```

Let me push that change.. good spot :)

Reply:
digging into the failing unit test, it appears this does need to be `*v = bytes`, which passes both the unit and the acctest at our end 🤷‍♂

```
$ envchain azurerm go test -v ./storage/2018-11-09/file/files/ -run="TestPutSmallFile"
=== RUN   TestPutSmallFile
2019/06/19 19:21:04 Testing if Service Principal / Client Certificate is applicable for Authentication..
2019/06/19 19:21:04 Testing if Service Principal / Client Secret is applicable for Authentication..
2019/06/19 19:21:04 Using Service Principal / Client Secret for Authentication
2019/06/19 19:21:32 [DEBUG] Chunk 1 of 1
--- PASS: TestPutSmallFile (75.26s)
    range_put_file_test.go:64: [DEBUG] Creating Top Level File..
    range_put_file_test.go:73: [DEBUG] Uploading File..
    range_put_file_test.go:78: [DEBUG] Deleting Top Level File..
PASS
```

```
=== RUN   TestByUnmarshallingBytes
--- PASS: TestByUnmarshallingBytes (0.00s)
PASS

Process finished with exit code 0
```

Reply:
Yes I was thinking about this after looking at this change.  The pointer-to-slice is passed by value, so when you dereference it you can change what it points to.